packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -264,10 +281,12 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
               margin: _isChildExpanded(index) ? kExpandedEdgeInsets : EdgeInsets.zero,
               child: ConstrainedBox(
                 constraints: const BoxConstraints(minHeight: _kPanelHeaderCollapsedHeight),
-                child: child.headerBuilder(
-                  context,
-                  _isChildExpanded(index),
-                ),
+                child: child.canTapOnHeader
```

Comment:
I think this implementation could look somewhat unusual, since the `Inkwell` doesn't encompass the `ExpandIcon` button. My suggestion would be to implement something along the lines of the following:
```
      final Row header = Row(
        children: <Widget>[
          Expanded(
            child: AnimatedContainer(
              duration: widget.animationDuration,
              curve: Curves.fastOutSlowIn,
              margin: _isChildExpanded(index) ? kExpandedEdgeInsets : EdgeInsets.zero,
              child: ConstrainedBox(
                constraints: const BoxConstraints(minHeight: _kPanelHeaderCollapsedHeight),
                child: headerWidget,
              ),
            ),
          ),
          Container(
            margin: const EdgeInsetsDirectional.only(end: 8.0),
            child: ExpandIcon(
              isExpanded: _isChildExpanded(index),
              padding: const EdgeInsets.all(16.0),
              onPressed: !child.canTapOnHeader // if canTapOnHeader is true, disable button
                ? (bool isExpanded) => _handlePressed(isExpanded, index)                
                : null,
            ),
          ),
        ],
      );

      items.add(
        MaterialSlice(
          key: _SaltedKey<BuildContext, int>(context, index * 2),
          child: Column(
            children: <Widget>[
              MergeSemantics(
                child: child.canTapOnHeader // if canTapOnHeader is true, wrap entire header in Inkwell and add tap behavior
                  ? InkWell(
                      onTap: () => _handlePressed(_isChildExpanded(index), index),
                      child: header,
                    )
                  : header,
              ),
    ... rest of code
```

Reply:
Let me know if you have any apprehensions to this approach, but I believe this would have a cleaner feel

Reply:
@shihaohong yes including the `ExpandIcon` does feel more 'natural', thanks!

Reply:
I would add tests to:
1) Verify that when `canTapOnHeader` is set to `false`, tapping on the rest of the header header shouldn't cause the panels to expand.
2) (unsure how I feel on this one, but you can decide) Ensure that `canTapOnHeader` is set to `false` by default and verify that behavior to be true.

Otherwise, this looks good!

Reply:
Thank you for the help! 

I have updated the tests accordingly, please take a look.